content/how-to/author-libraries.md

```diff
@@ -1,7 +1,207 @@
+We will add a `babel-loader` as our library is written in es2015 syntax. Add preset [`babel-preset-es2015`](https://www.npmjs.com/package/babel-preset-es2015) to transpile es2015 to es5. But for webpack2 we can disable `modules` as webpack2 understands es2015 module system.
+Similarly a `json-loader` is required to precompile our json fixture file.
+
+```javascript
+export default () => (
+    {
+        ...
+        module: {
+            rules: [{
+                test: /.js$/,
+                exclude: /node_modules/,
+                use: 'babel-loader',
+                options: {
+                    presets: [
+                        ["es2015", { "modules": false }]
+                    ]
+                }
+            }, {
+                test: /.json$/,
+                exclude: /node_modules/,
+                use: 'json-loader'
+            }]
+        }
+    }
+);
+```
+### Add `externals`
+
+Now, if you run `webpack`, you will find that a largish bundle file is created. If you inspect the file, you will find that lodash has been bundled along with your code.
+It would be unnecessary for your library to bundle a library like `lodash`. Hence you would want to give up control of this external library to the consumer of your library.
+
+This can be done using the `externals` configuration as
+
+```javascript
+export.default () => ({
+    ...
+    externals: [{
+        '_' : 'lodash'
+    }]
+    ...
+})
+```
+
+This means that your library expects `lodash` to be available in the consumers environment, with a global name of `_`. It essentially aliases `_` to the library `lodash`.
+Also this configuration prevents webpack from bundling this external library to your library code.
+
+### Add `libraryTarget`
+
+For widespread use of the library, we would like it to be compatible in different environments, ie CommonJs, AMD, nodejs and as a global variable.
+
+To make your library available for reuse, add `library` property in webpack configuration.
+
+```javascript
+export default = () => ({
+    ...
+    output: {
+        ...
+        library: 'webpackNumbers'
+    }
+    ...
+})
+```
+
+This makes your library bundle to be available as a global variable when imported.
+To make the library compatible with other environments, add `libraryTarget` property to the config.
+
+```javascript
+export default = () => ({
+    ...
+    output: {
+        ...
+        library: 'webpackNumbers',
+        libraryTarget:'umd' // Possible value - amd, commonjs, commonjs2, commonjs-module, this, var
+    }
+    ...
+})
+```
+
+If `library` property is set and `libraryTarget` is set to be `var` by default, as given in the [config reference](/configuration/output).
+
+### Final Steps
+
+[Tweak your production build using webpack](/how-to/generate-production-build).
+
+Add the path to your generated bundle as the package's main file in `package.json`
+
+```json
+{
+    ...
+    "main": "dist/webpack-numbers.js"
```

Comment:
Sorry again! I couldn't find it in the [documentation](https://docs.npmjs.com/files/package.json). Does `package.json` have a module field?